cmd/handler-utils.go

```diff
@@ -161,25 +161,32 @@ func extractMetadata(ctx context.Context, r *http.Request) (metadata map[string]
 }
 
 // extractMetadata extracts metadata from map values.
-func extractMetadataFromMap(ctx context.Context, v map[string][]string, m map[string]string) error {
+func extractMetadataFromMime(ctx context.Context, v textproto.MIMEHeader, m map[string]string) error {
 	if v == nil {
 		logger.LogIf(ctx, errInvalidArgument)
 		return errInvalidArgument
 	}
+
+	nv := make(textproto.MIMEHeader, len(v))
+	for k, kv := range v {
+		// Canonicalize all headers, to remove any duplicates.
+		nv[http.CanonicalHeaderKey(k)] = kv
```

Comment:
Should we skip canonicalization for user set metadata? `X-Amz-Meta-Documenttype` displayed as `x-amz-meta-documenttype` on HEAD

Reply:
no @poornas this is done on purpose so that we don't cause confusion when we grab it from nv.Get() because if its not canonicalized `nv.Get()` never returns lowercased header.